src/box/lua/iproto.c

```diff
@@ -287,23 +168,38 @@ push_iproto_protocol_features(struct lua_State *L)
 	lua_pushinteger(L, IPROTO_CURRENT_VERSION);
 	lua_setfield(L, -2, "protocol_version");
 
-	const struct iproto_constant features[] = {
-		{"streams", IPROTO_FEATURE_STREAMS},
-        	{"transactions", IPROTO_FEATURE_TRANSACTIONS},
-        	{"error_extension", IPROTO_FEATURE_ERROR_EXTENSION},
-        	{"watchers", IPROTO_FEATURE_WATCHERS},
-        	{"pagination", IPROTO_FEATURE_PAGINATION},
-	};
+	size_t region_svp = region_used(&fiber()->gc);
+	char **names = xregion_alloc_array(
+		&fiber()->gc, char *,
+		iproto_feature_id_constants_size
+	);
+	struct iproto_constant *features = xregion_alloc_array(
+		&fiber()->gc, struct iproto_constant,
+		iproto_feature_id_constants_size
+	);
+	const char *prefix = "IPROTO_FEATURE_";
+	size_t prefix_len = strlen(prefix);
+	for (size_t i = 0; i < iproto_feature_id_constants_size; ++i) {
+		const char *name = iproto_feature_id_constants[i].name;
+		assert(strncmp(name, prefix, prefix_len) == 0);
+		names[i] = strtolowerdup(name + prefix_len);
+		features[i].name = names[i];
```

Comment:
Looks like `names` is redundant. Anyway, preparing the features array on the region looks ugly. AFAIU you need it only to convert feature names to lowercase. If so, better add a flag to `push_iproto_constant_subnamespace` that would do it instead of allocating a temporary array on the region.